src/emitters/astroobject.cpp

```diff
@@ -0,0 +1,256 @@
+#include <mitsuba/core/bsphere.h>
+#include <mitsuba/core/properties.h>
+#include <mitsuba/core/warp.h>
+#include <mitsuba/render/emitter.h>
+#include <mitsuba/render/scene.h>
+#include <mitsuba/render/texture.h>
+
+NAMESPACE_BEGIN(mitsuba)
+
+/**!
+
+.. _emitter-astroobject:
+
+Distant astroobject emitter (:monosp:`astroobject`)
+---------------------------------------------------
+
+.. pluginparameters::
+
+ * - irradiance
+   - |spectrum|
+   - Spectral irradiance, which corresponds to the amount of spectral power
+     per unit area received by a hypothetical surface normal to the specified
+     direction.
+   - |exposed|, |differentiable|
+
+ * - to_world
+   - |transform|
+   - Emitter-to-world transformation matrix.
+   - |exposed|
+
+ * - direction
+   - |vector|
+   - Alternative (and exclusive) to `to_world`. Direction towards which the
+     emitter is radiating in world coordinates.
+
+ * - angular_diameter
+   - |float|
+   - Angular diameter of the astroobject in degrees.
+
+This emitter plugin implements a distant astroobject source which radiates a
+specified power per unit area within a solid angle. This solid angle is the
+proportion of the FOV taken by the emitter. This is parameterized by an
+angular_diameter. By default, the emitter radiates in the direction of the
+positive Z axis, i.e. :math:`(0, 0, 1)`, with an angular diameter of about the
+equivalent average sun angular diameter from earth.
+
+.. tabs::
+    .. code-tab:: xml
+        :name: astroobject-light
+
+        <emitter type="astroobject">
+            <vector name="direction" value="1.0, 0.0, 0.0"/>
+            <rgb name="irradiance" value="1.0"/>
+        </emitter>
+
+    .. code-tab:: python
+
+        'type': 'astroobject',
+        'direction': [1.0, 0.0, 0.0],
+        'irradiance': {
+            'type': 'rgb',
+            'value': 1.0,
+        }
+
+*/
+
+#define SUN_ANGULAR_DIAMETER 0.5358f
```

Comment:
Would you mind adding a little comment explaining briefly how you came up with this value?

Reply:
I took the value from [here](https://github.com/mitsuba-renderer/mitsuba/blob/master/src/emitters/sun.cpp). I may add the comment in the astroobject code